checkpoint/recovery.go

```diff
@@ -216,65 +219,149 @@ type recoverydata struct {
 	atxs     []*atxs.CheckpointAtx
 }
 
-type ownAtxData struct {
-	atx      *types.VerifiedActivationTx
-	preserve []*types.VerifiedActivationTx
-	proof    []byte
-}
-
 func preserveOwnData(
+	logger log.Log,
 	db *sql.Database,
 	cfg *RecoverConfig,
 	nodeID types.NodeID,
 	data *recoverydata,
-) (ownAtxData, error) {
-	var own ownAtxData
+) ([]*types.VerifiedActivationTx, [][]byte, error) {
 	if !cfg.PreserveOwnAtx {
-		return own, nil
+		return nil, nil, nil
 	}
 	atxid, err := atxs.GetLastIDByNodeID(db, nodeID)
 	if err != nil && !errors.Is(err, sql.ErrNotFound) {
-		return own, fmt.Errorf("query own last atx id: %w", err)
+		return nil, nil, fmt.Errorf("query own last atx id: %w", err)
 	}
+	var seed types.ATXID
+	var own bool
 	if atxid == types.EmptyATXID {
-		return own, nil
+		// check if any atx is being built
+		if cfg.PostDataDir == "" {
+			return nil, nil, nil
+		}
+		m, err := initialization.LoadMetadata(cfg.PostDataDir)
+		if err != nil {
+			return nil, nil, fmt.Errorf("read post metdata %s: %w", cfg.PostDataDir, err)
+		}
+		seed = types.ATXID(types.BytesToHash(m.CommitmentAtxId))
+		logger.With().Debug("found commitment atx from metadata", seed)
+	} else {
+		seed = atxid
+		logger.With().Debug("found own atx", seed)
```

Comment:
named fields (e.g when thing is passed to the logger without name) have to be deprecated eventually